scylla/src/client/session.rs

```diff
@@ -902,6 +902,7 @@ impl Session {
             connection_config,
             pool_size: config.connection_pool_size,
             can_use_shard_aware_port: !config.disallow_shard_aware_port,
+            connectivity_events_sender: None, // This will be set by `Cluster::new`.
         };
 
```

Comment:
This becomes yet another way signal out of connection pool - we already had `pool_updated_notify` and `pool_empty_notifier`. Instead of introducing third one, can we do some unification?

`pool_empty_notifier` sounds like something really close to what we need - its just that we need to send both when it becomes empty, and starts being empty. Current users of `pool_empty_notifier` could ignore the "stops being empty" case.

Reply:
As discussed, possible unification of these notification channels is out of scope of this PR.